src/index.html

```diff
@@ -11,7 +12,155 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
+    <form action="catalog" method="post">
+      <fieldset>
+        <legend>Personal information</legend>
+
+        
+        <label class="form-field">
+          Surname:
+          <input 
+            type="text" 
+            name="surname" 
+            required
+          >
+        </label>
+        <label class="form-field">
+          Name:
+          <input 
+            type="text" 
+            name="name" 
+            required
+          >
+        </label>
+        <label class="form-field">
+          How old are you?
+          <input 
+            type="number" 
+            name="howoldareyou" 
+            value="12"
+            min="1"
+            max="100" 
+            required
+          >
+        </label>
+        <label class="form-field">
+          Full date of birth:
+          <input 
+            type="date" 
+            name="fulldateofbirth" 
+            placeholder="mm/dd/yyyy" 
+            required
+          >
+        </label>
+        <label class="form-field">
+          I accept the term of the agreement
+          <input 
+            type="checkbox" 
+            name="iacceptthetermoftheagreement" 
+            required
+          >
+        </label> 
+      </fieldset>     
+      <fieldset>
+        <legend>Registration</legend>
+
+        <label class="form-field">
+          E-mail:
+          <input 
+            type="email" 
+            name="e-mail" 
+            placeholder="email@example.com" 
+            required
+          >
+        </label>
+        <label class="form-field">
+          Password:
+          <input 
+            type="password" 
+            name="password" 
+            required
+            minlength="5"
+            maxlength="32"
+          >
+        </label>
+      </fieldset>
+      <fieldset>
+        <legend>An interesting fact about you!</legend>
+        <div class="form-field">
+            Do you love cats?
+
+          <label>
+            <input 
+              type="radio" 
+              name="cat" 
+              value="y"
+            >
+            Yes
+          </label>
+
+          <label>
+            <input 
+              type="radio" 
+              name="cat" 
+              value="n"
+            >
+            No
+          </label>
+        </div>
+        <label class="form-field">
+            What is your favorite color?
+          <input 
+            type="color" 
+            name="whatisyourfavoritecolor?"
+          >
+        </label>
```

Comment:
```suggestion
        <label class="form-field">
          What is your favorite color?
          <input 
            type="color" 
            name="whatisyourfavoritecolor?"
          >
        </label>
```
text is on the same nesting level as input

Reply:
Didn't fixed